git_machete/cmd.py

```diff
@@ -766,14 +1632,14 @@ def find_executable(cli_ctxt: CommandLineContext, executable: str) -> Optional[s
     return None
 
 
-def get_default_editor(cli_ctxt: CommandLineContext) -> str:
+def get_default_editor(machete_client: MacheteClient) -> str:
```

Comment:
Hmmm I think passing just `cli_ctxt` is sufficient here? 🤔

Reply:
As in, this method doesn't seem to rely on anything specific to `MacheteClient`

Reply:
MacheteClient object is used here to pass definition_file_path

Reply:
Hmmm I'd refactor this piece so that `get_default_editor` doesn't need to refer to `definition_file_path`: make `get_default_editor` return `None` if nothing is found, and let the call site (`edit()` method) raise an error instead

Reply:
(since the only place where `definition_file_path` is needed is in the error message)